pkg/scheduler/cache/event_handlers.go

```diff
@@ -49,8 +49,8 @@ func (sc *SchedulerCache) getOrCreateJob(pi *schedulingapi.TaskInfo) *scheduling
 		if pi.Pod.Spec.SchedulerName != sc.schedulerName {
 			klog.V(4).Infof("Pod %s/%s will not scheduled by %s, skip creating PodGroup and Job for it",
 				pi.Pod.Namespace, pi.Pod.Name, sc.schedulerName)
+			return nil
 		}
-		return nil
```

Comment:
If `pi.Job` is empty, it should also return nil :)

Reply:
Oops, I misunderstood the comment.